enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,107 @@
+
+This enhancement aims to provide a robust solution for managing certificates for the API endpoint in baremetal environments.
+
+### User Stories
+
+1. **As a cluster administrator**, I want to issue custom certificates for the API endpoint (`api.cluster.example.com`) using an external ACME CA, so that I can ensure secure communication for my cluster's API.
+2. **As a cluster administrator on a baremetal platform**, I want a reliable solution to handle HTTP01 challenges for the API endpoint, even when the endpoint is not managed by OpenShift Ingress, so that I can avoid manual workarounds.
+3. **As a developer**, I want a simple deployment mechanism for the HTTP01 challenge proxy, so that I can easily integrate it into my existing cluster setup.
+
+### Proposal
+
+The HTTP01 Challenge Proxy will be implemented via DaemonSet running on the cluster. It will:
+
+- Redirect HTTP traffic from the API endpoint (`api.cluster.example.com`) on port 80 to the OpenShift Ingress Routers.
+- Use `nftables` for traffic redirection from `API:80` to `PROXY:8888`.
+- Be deployed using a manifest that includes all necessary configurations.
+
+The proxy will ensure compatibility with various OCP topologies, including SNO, MNO, and Compact clusters, addressing the challenges of HTTP01 validation for the API endpoint.
+
+### API Extensions
+
+A new CR type may be created and can be applied to clusters.  This new typed will be stored in the [openshift/api](https://github.com/openshift/api) repo.
+
+### Implementation Details/Notes/Constraints
+
+- The proxy will be deployed as a DaemonSet to ensure it runs on all nodes in the cluster.
+- The nftables rules will be added to the nodes. The proxy will listen on port 8888 and redirect traffic to the OpenShift Ingress Routers.
+- The implementation relies on `nftables` for traffic redirection, which must be supported and enabled on the cluster nodes.
+- The demo deployment manifest for the proxy is available [here](https://github.com/mvazquezc/cert-mgr-http01-proxy/blob/main/manifests/deploy-in-ocp.yaml).
+- An example implementation can be found in this [repository](https://github.com/mvazquezc/cert-mgr-http01-proxy/tree/main).
+
+### Design Details
+
+- **Proxy Deployment**: The proxy will be deployed using a Kubernetes DaemonSet. The daemonset will implement an nftable rule via pod that runs to completion.
+- **Traffic Redirection**: This will use `nftables` rules to redirect incoming traffic on `API:80` to `PROXY:8888`.
+- **Security**: The proxy will only handle HTTP traffic for the HTTP01 challenge and will not interfere with other traffic or services.
+- **Monitoring**: Logs and metrics will be exposed to help administrators monitor the proxy's behavior and troubleshoot issues.
+
+### Drawbacks
+
+1. **Dependency on nftables**: The solution relies on `nftables`, which may not be available or enabled on all environments.
```

Comment:
nftables is available on control plane (and worker) nodes.

Reply:
It is _always_ available? What is the expected behavior if it is not available but someone attempts to use this feature?

Reply:
Might be a better question for @imiller0 or @mvazquezc if the nftables doesn't exist.  I can edit the enhancement proposal with what is discussed.

Reply:
If I'm not mistaken, the nftables are always available. May be worth double-checking with SDN team but iirc the SDN relies on nftables as well.